src/StatusQ/Controls/StatusBaseButton.qml

```diff
@@ -19,158 +20,106 @@ Rectangle {
         Primary
     }
 
-    property StatusAssetSettings icon: StatusAssetSettings {
-        width: 24
-        height: 24
-    }
+    property StatusAssetSettings asset: StatusAssetSettings { }
 
     property bool loading: false
 
-    property alias hovered: sensor.containsMouse
+    property color normalColor
+    property color hoverColor
+    property color disabledColor
+
+    property color textColor
+    property color disabledTextColor
+    property color borderColor: "transparent"
 
     property int size: StatusBaseButton.Size.Large
     property int type: StatusBaseButton.Type.Normal
 
-    property alias text: label.text
-    property alias font: label.font
-
-    property real defaultLeftPadding: size === StatusBaseButton.Size.Large ? 24 : 12
-    property real defaultRightPadding: size === StatusBaseButton.Size.Large ? 24 : 12
-    property real defaultTopPadding: {
-        switch (size) {
-            case StatusBaseButton.Size.Tiny:
-              return 5
-            case StatusBaseButton.Size.Small:
-              return 10
-            case StatusBaseButton.Size.Large:
-            default:
-              return 11
-        }
-    }
-    property real defaultBottomPadding: {
-        switch (size) {
+    QtObject {
+        id: d
+        readonly property color textColor: root.enabled ? root.textColor : root.disabledTextColor
+        property int defaultHorizontalPadding: (((size === StatusBaseButton.Size.Large) ? 48 : 24) + root.horizontalPadding)
+        property int defaultVerticalPadding: {
```

Comment:
No :) You don't need these `defaultHorizontalPadding/defaultVerticalPadding` here in the `d` object, just put them at the root level:

```qml
Button {
   horizontalPadding: foo
   verticalPadding: bar
```

Reply:
the only thing I see if we do this, is that if user sets for example horizontal and vertical padding to 10, this is what they get:
<img width="115" alt="f" src="https://user-images.githubusercontent.com/31625338/189118273-bfd459f2-c255-4d9d-a2b1-258fe847de7b.png">
So I guess it's not bad to keep some default (and fix) sizes, right?

Reply:
It's because your `contentItem` doesn't have an implicitSize (you're just using a plain Item with a Row inside which has `anchors.centerIn`).